src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -1405,8 +1409,138 @@ int64_t SystemNative_GetFileSystemType(intptr_t fd)
+    else if (strcmp(statfsArgs.f_basetype, "qnx6") == 0) result = 0x68191122;
+    else if (strcmp(statfsArgs.f_basetype, "ramfs") == 0) result = 0x858458F6;
+    else if (strcmp(statfsArgs.f_basetype, "reiserfs") == 0) result = 0x52654973;
+    else if (strcmp(statfsArgs.f_basetype, "romfs") == 0) result = 0x7275;
+    else if (strcmp(statfsArgs.f_basetype, "rootfs") == 0) result = 0x53464846;
+    else if (strcmp(statfsArgs.f_basetype, "rpc_pipefs") == 0) result = 0x67596969;
+    else if (strcmp(statfsArgs.f_basetype, "samba") == 0) result = 0x517B;
+    else if (strcmp(statfsArgs.f_basetype, "securityfs") == 0) result = 0x73636673;
+    else if (strcmp(statfsArgs.f_basetype, "selinux") == 0) result = 0xF97CFF8C;
+    else if (strcmp(statfsArgs.f_basetype, "sffs") == 0) result = 0x786F4256;
+    else if (strcmp(statfsArgs.f_basetype, "sharefs") == 0) result = 0x01021994;
+    else if (strcmp(statfsArgs.f_basetype, "smb") == 0) result = 0x517B;
+    else if (strcmp(statfsArgs.f_basetype, "smb2") == 0) result = 0xFE534D42;
+    else if (strcmp(statfsArgs.f_basetype, "sockfs") == 0) result = 0x534F434B;
+    else if (strcmp(statfsArgs.f_basetype, "squashfs") == 0) result = 0x73717368;
+    else if (strcmp(statfsArgs.f_basetype, "sysfs") == 0) result = 0x62656572;
+    else if (strcmp(statfsArgs.f_basetype, "sysv2") == 0) result = 0x012FF7B6;
+    else if (strcmp(statfsArgs.f_basetype, "sysv4") == 0) result = 0x012FF7B5;
+    else if (strcmp(statfsArgs.f_basetype, "tmpfs") == 0) result = 0x01021994;
+    else if (strcmp(statfsArgs.f_basetype, "ubifs") == 0) result = 0x24051905;
+    else if (strcmp(statfsArgs.f_basetype, "udf") == 0) result = 0x15013346;
+    else if (strcmp(statfsArgs.f_basetype, "ufs") == 0) result = 0x00011954;
+    else if (strcmp(statfsArgs.f_basetype, "ufscigam") == 0) result = 0x54190100;
+    else if (strcmp(statfsArgs.f_basetype, "ufs2") == 0) result = 0x19540119;
+    else if (strcmp(statfsArgs.f_basetype, "usbdevice") == 0) result = 0x9FA2;
+    else if (strcmp(statfsArgs.f_basetype, "v9fs") == 0) result = 0x01021997;
+    else if (strcmp(statfsArgs.f_basetype, "vagrant") == 0) result = 0x786F4256;
+    else if (strcmp(statfsArgs.f_basetype, "vboxfs") == 0) result = 0x786F4256;
+    else if (strcmp(statfsArgs.f_basetype, "vmhgfs") == 0) result = 0xBACBACBC;
+    else if (strcmp(statfsArgs.f_basetype, "vxfs") == 0) result = 0xA501FCF5;
+    else if (strcmp(statfsArgs.f_basetype, "vzfs") == 0) result = 0x565A4653;
+    else if (strcmp(statfsArgs.f_basetype, "xenfs") == 0) result = 0xABBA1974;
+    else if (strcmp(statfsArgs.f_basetype, "xenix") == 0) result = 0x012FF7B4;
+    else if (strcmp(statfsArgs.f_basetype, "xfs") == 0) result = 0x58465342;
+    else if (strcmp(statfsArgs.f_basetype, "xia") == 0) result = 0x012FD16D;
+    else if (strcmp(statfsArgs.f_basetype, "udev") == 0) result = 0x01021994;
+    else if (strcmp(statfsArgs.f_basetype, "zfs") == 0) result = 0x2FC12FC1;
+
```

Comment:
it would be nice to assert the result here to detect missing mappings (might happen in a few years from now):

```suggestion

    assert(result != -1);
```

Ideally we would include `f_basetype` in the error msg, but I am not sure if the following code is going to compile:

```c
    assert_msg(result != -1, "Unknown file system type", statfsArgs.f_basetype);
```

Reply:
Unfortunately, `assert_msg` only accepts `int` as third argument.

It would be good to add have string variant of that macro (or change existing `assert_*`'s to convert `int`s to string and accept only string). Maybe at the same time, we should also move `assert_*` and `c_static_assert_*` macros in a single header (rather than split them in `pal_utilities.c` and `pal_compiler.c`).